internal/adapter/adapter.go

```diff
@@ -68,24 +68,32 @@ func (b *Bot) GetFirstName() string {
 
 func (b *Bot) adaptUpdates(updates <-chan tgbotapi.Update, messages chan<- *Message) {
 	var err error
+	var msg *tgbotapi.Message
 	for update := range updates {
+		if update.Message != nil {
```

Comment:
update.Message and update.ChannelPost could be nil simultaneously (in case it's edit event, not new message)